core/arch/arm/mm/tee_mm.c

```diff
@@ -179,7 +183,7 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, size_t size)
 			 */
 			if ((entry->offset << pool->shift) < size)
 				/* out of memory */
-				return NULL;
+				goto err;
```

Comment:
even if this is a bit outside to the scope of the change, please add braces around this `if`/`then`. It is not nice separating a one line instruction from its related if/while/do. I mean I would really prefer:
```c
if (...) {
    /* some comment */
    some_single_instruction;
}
```



Same below around line 196.